src/Orchard/Security/IMembershipSettings.cs

```diff
@@ -21,5 +21,13 @@ public interface IMembershipSettings {
         MembershipPasswordFormat PasswordFormat { get; set; }
         bool EnablePasswordHistoryPolicy { get; set; }
         int PasswordReuseLimit { get; set; }
+        bool EnableCustomUsernamePolicy { get; set; }
+        int MinimumUsernameLength { get; set; }
+        int MaximumUsernameLength { get; set; }
+        bool ForbidUsernameSpecialChars { get; set; }
+        bool AllowEmailAsUsername {get; set;}
```

Comment:
This is introducing a security issue if the login allows for both, you could impersonate someone else (we changed it in OC).

Reply:
To avoid that, if the username is an email we check it's the same email of the user.

For reference: 

In the UserService class

`if (usernameIsEmail && !username.Equals(email, StringComparison.OrdinalIgnoreCase)){
                validationErrors.Add(new UsernameValidationError(Severity.Fatal, UsernameValidationResults.UsernameAndEmailMustMatch,
                        T("If the username is an email it must match the specified email address.")));
                return false;
            }
`

Reply:
Do you check the unicity of a new username to see if it's not an existing email address?

Reply:
Yes. I double checked and found a case where the backoffice user could change the username to an. already existing username and I fixed it. In every other case it was already checked